src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -345,10 +346,10 @@ trait Infer extends Checkable {
      *  by existentially bound variables.
      */
     def makeFullyDefined(tp: Type): Type = {
-      var tparams: List[Symbol] = Nil
+      val tparams: ListBuffer[Symbol] = ListBuffer.empty
```

Comment:
I wonder what's the distribution of `tparams.length` at the end of this method? If `0` is the most common case, this change will be counter-productive.

You could leave this as `null` to start with, and allocate lazily on for first addition of a `tparam`.

Reply:
Done!